esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -0,0 +1,373 @@
+    // otherise => increment so that the previously return scopes are not returned
+    minVersion = reqMinVersion ? reqMinVersion+1 : 0;
+
+    unsigned __int64 maxTimestamp = 0;
+    IArrayOf<IEspWUScope> respScopes;
+    if (scopeTypes.ordinality()>0)
+    {
+        if (scopes.ordinality()>0)
+            throw MakeStringException(ECLWATCH_INVALID_INPUT, "ScopeType filter may not be used with scope filter");
+        if (ids.ordinality()>0)
+            throw MakeStringException(ECLWATCH_INVALID_INPUT, "ScopeType filter may not be used with id filter");
+
+        DBGLOG("onWUDetails: Filter by ScopeType");
+        nestedDepth = nfilter.getDepth();
+        if (nestedDepth)
+        {
+            buildScopeTypeListFromStringArray(scopeTypes,scopeTypesFilterList);
+            wustatsfilter->setScopeDepth(0, scopeFilter.getMaxDepth()+nfilter.getDepth());
+            Owned<IConstWUStatisticIterator> stats = & workunit->getStatistics(wustatsfilter);
+            processStatsIntoResponse(respScopes, *stats, maxTimestamp);
+        }
+        else
+        {
+            wustatsfilter->setScopeDepth(0, scopeFilter.getMaxDepth());
+            ForEachItemIn(idx, scopeTypes)
+            {
+                StatisticScopeType scopeType = queryScopeType(scopeTypes[idx]);
+                if (scopeType == SSTnone)
+                    throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid ScopeType: %s", scopeTypes[idx]);
+                wustatsfilter->setScopeType(scopeType);
+
+                Owned<IConstWUStatisticIterator> stats = & workunit->getStatistics(wustatsfilter);
+                processStatsIntoResponse(respScopes, *stats, maxTimestamp);
+            }
+        }
+    }
+    else if (scopes.ordinality()>0)
+    {
+        DBGLOG("onWUDetails: Filter by Scope");
+        nestedDepth = 0;
+        wustatsfilter->setScopeDepth(0, scopeFilter.getMaxDepth());
+        ForEachItemIn(idx, scopes)
+        {
+            if(nfilter.getDepth())
+            {
+                StringBuffer scopeMatch(scopes[idx]);
+                scopeMatch.append("*");
+                wustatsfilter->setScope(scopeMatch);
+            }
+            else
+            {
+                wustatsfilter->setScope(scopes[idx]);
+            }
+
+            Owned<IConstWUStatisticIterator> stats = & workunit->getStatistics(wustatsfilter);
+            processStatsIntoResponse(respScopes, *stats, maxTimestamp);
+        }
+    }
+    else
+    {
+        nestedDepth = nfilter.getDepth();
+        wustatsfilter->setScopeDepth(0, scopeFilter.getMaxDepth()+nfilter.getDepth());
+        DBGLOG("onWUDetails: Filter by id");
+        Owned<IConstWUStatisticIterator> stats = & workunit->getStatistics(wustatsfilter);
+        processStatsIntoResponse(respScopes, *stats, maxTimestamp);
+    }
+    StringBuffer maxVersion;
+    maxVersion.append(maxTimestamp);
+
+    resp.setWUID(wuid.str());
+    resp.setMaxVersion(maxVersion.str());
```

Comment:
@ghalliday ESP uses string type for int64 ecm data types.  (It does use integers for int32).